pkg/hostagent/ipam.go

```diff
@@ -37,18 +36,35 @@ func combine(ranges []*ipam.IpAlloc) *ipam.IpAlloc {
 	return result
 }
 
-// must have index lock
-func (agent *HostAgent) rebuildIpam() {
+// builds the used IP info from metadata, at init.
+func (agent *HostAgent) buildUsedIPs() {
```

Comment:
No locking is needed to protect the map used in this method?

Reply:
This is only executed at start up to build the map from the ep files. At that time, the agent is not yet running. Thus, lock is not necessary.